src/main/java/com/uid2/admin/vertx/service/OperatorKeyService.java

```diff
@@ -144,22 +169,48 @@ private void handleOperatorAdd(RoutingContext rc) {
             // refresh manually
             operatorKeyProvider.loadContent(operatorKeyProvider.getMetadata());
 
+            if (!rc.queryParams().contains("name")) {
+                ResponseUtil.error(rc, 400, "no name specified");
+                return;
+            }
             final String name = rc.queryParam("name").get(0);
-            Optional<OperatorKey> existingOperator = this.operatorKeyProvider.getAll()
+
+            final Optional<OperatorKey> existingOperator = this.operatorKeyProvider.getAll()
                     .stream().filter(o -> o.getName().equals(name))
                     .findFirst();
             if (existingOperator.isPresent()) {
                 ResponseUtil.error(rc, 400, "key existed");
                 return;
             }
 
-            String protocol = RequestUtil.validateOperatorProtocol(rc.queryParam("protocol").get(0));
+            final String protocol = rc.queryParams().contains("protocol")
+                    ? RequestUtil.validateOperatorProtocol(rc.queryParam("protocol").get(0))
+                    : null;
             if (protocol == null) {
                 ResponseUtil.error(rc, 400, "no protocol specified");
                 return;
             }
 
-            List<OperatorKey> operators = this.operatorKeyProvider.getAll()
+            Set<Role> roles = RequestUtil.getRoles(rc.queryParam("roles").get(0));
```

Comment:
`rc.queryParam("roles").get(0)` may throw an NPE especially if you want to support backwards compatibility.

It would be better to do `rc.queryParam("roles") == null ? Set.of() : rc.queryParam("roles").get(0)`

Reply:
Also, you need to validate whether all roles are not null